### ISSUE TITLE
email validation failed while input is ok

### ISSUE THREAD

**Issue by GiantCrocodile** [sailsjs/anchor#94]
See this output:
```
    "email": [
      {
        "rule": "email",
        "message": "\"email\" validation rule failed for input: 'bla@domain.com'"
      }
    ],
```

**Comment by particlebanana** [comment]
Passes through to https://github.com/chriso/validator.js/blob/master/validator.js#L174

**Comment by GiantCrocodile** [comment]
You mean I should use isEmail as validation rule name @particlebanana ? That doesn't work.

**Comment by particlebanana** [comment]
No if there are any issues with the validation you can make an issue on that repo.

Does it pass for any email addresses?

**Comment by GiantCrocodile** [comment]
It fails for any email address I provide @particlebanana.

**Comment by sylvainlap** [comment]
I got the same error:
```
"email": [
      {
        "rule": "email",
        "message": "\"email\" validation rule failed for input: 'sylvain@gmail.com'"
      }
    ]
```
the validation fail for every email address.

**Comment by GiantCrocodile** [comment]
That's why I think this problem is clearly a problem of anchor and not of validator. I doubt that validator is failing every email but anchor implemented it wrongly.

**Comment by jeff-kilbride** [comment]
I am also having this problem with validation on email addresses. The error is being thrown because validator is trying to assign a property to a boolean primitive on line 692 of validator.js. Here is the stack trace of the actual error:

```javascript
TypeError: Cannot assign to read only property 'allow_display_name' of true
    at merge (/usr/lib/node_modules/sails/node_modules/anchor/node_modules/validator/validator.js:692:26)
    at Object.validator.isEmail (/usr/lib/node_modules/sails/node_modules/anchor/node_modules/validator/validator.js:168:19)
    at Object.validator.extend.validator.(anonymous function) (/usr/lib/node_modules/sails/node_modules/anchor/node_modules/validator/validator.js:96:23)
    at Object.matchRule (/usr/lib/node_modules/sails/node_modules/anchor/lib/match/matchRule.js:40:20)
    at Anchor.to (/usr/lib/node_modules/sails/node_modules/anchor/index.js:76:45)
    at /usr/lib/node_modules/sails/node_modules/waterline/lib/waterline/core/validations.js:188:31
    at _toString (/usr/lib/node_modules/sails/node_modules/waterline/node_modules/async/lib/async.js:49:16)
    at done (/usr/lib/node_modules/sails/node_modules/waterline/node_modules/async/lib/async.js:239:19)
    at called (/usr/lib/node_modules/sails/node_modules/waterline/node_modules/async/lib/async.js:40:16)
    at /usr/lib/node_modules/sails/node_modules/waterline/lib/waterline/core/validations.js:164:64
    at _setImmediate (/usr/lib/node_modules/sails/node_modules/waterline/node_modules/async/lib/async.js:162:20)
    at /usr/lib/node_modules/sails/node_modules/waterline/node_modules/async/lib/async.js:230:13
    at _arrayEach (/usr/lib/node_modules/sails/node_modules/waterline/node_modules/async/lib/async.js:81:9)
    at _each (/usr/lib/node_modules/sails/node_modules/waterline/node_modules/async/lib/async.js:72:13)
    at Object.async.forEachOf.async.eachOf (/usr/lib/node_modules/sails/node_modules/waterline/node_modules/async/lib/async.js:229:9)
    at Object.async.forEach.async.each (/usr/lib/node_modules/sails/node_modules/waterline/node_modules/async/lib/async.js:206:22)
```

The `validator.isEmail()` function expects a string and an options object. It is being passed a string (the email address) and a boolean (true). When it tries to merge the `default_email_options` with the wrongly passed boolean primitive (validator.js line 168), it blows up (validator.js line 692).  Anchor catches this and assumes it's a validation error (matchRule.js line 41).

I have a feeling this could be affecting other rules -- at least, any rules that rely on a non-object condition (e.g. true/false) that are merged with validator default rulesets. Maybe it's only the email rule that hits this sweet spot...  If I knew the code a little better, I would offer a PR, but I'm not really sure yet where the args being passed to `isEmail()` are originating from. I've just been dropping breakpoints and slowly trying to isolate the real error. 

And since this seems to be a bug, I wanted to report it sooner rather than later.

**Comment by particlebanana** [comment]
@jeff-kilbride this should be fixed in the latest published version of Anchor (0.11.0)
